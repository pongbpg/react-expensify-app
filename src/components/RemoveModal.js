import React from 'react';
import Modal from 'react-modal';

export default (props) => (
    <Modal
        isOpen={!!props.selectedRemove}
        contentLabel="Remove Expense"
        onRequestClose={props.handleCloseRemoveModal}
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Remove expense!</h3>
        {props.selectedRemove && <p className="modal__body">{props.selectedRemove}</p>}
        <div className="modal__button">
            <button className="button button--remove" onClick={props.handleConfirmRemoveModal}>Remove</button>
            <button className="button button--secondary" onClick={props.handleCloseRemoveModal}>Cancel</button>
        </div>
    </Modal>
);