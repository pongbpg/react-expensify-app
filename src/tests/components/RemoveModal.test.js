import React from 'react';
import { shallow } from 'enzyme';
import RemoveModal from '../../components/RemoveModal';

test('should render Remove Modal correctly', () => {
  const wrapper = shallow(
    <RemoveModal
      selectedRemove
      handleConfirmRemoveModal={() => {}}
      handleCloseRemoveModal={() => {}}
    />
  );
  expect(wrapper).toMatchSnapshot();
});