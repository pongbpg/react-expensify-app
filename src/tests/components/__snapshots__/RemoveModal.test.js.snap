// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Remove Modal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Remove Expense"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h3
    className="modal__title"
  >
    Remove expense!
  </h3>
  <p
    className="modal__body"
  />
  <div
    className="modal__button"
  >
    <button
      className="button button--remove"
      onClick={[Function]}
    >
      Remove
    </button>
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</Modal>
`;
